src/models/multifrequency.jl

```diff
@@ -0,0 +1,175 @@
+end
+
+#visibility_point(M::Multifrequency{B},p) where {B} = visibility_point(B)
+
+#"""
+#    $(TYPEDEF)
+#Taylor expansion spectral model of order n for multifrequency imaging. Applies to Continuous Image models only.
+#
+#This is the same multifrequency model implemented in ehtim (Chael et al., 2023).
+#
+# Fields
+#$(FIELDS)
+#"""
+#struct TaylorSpectral{C<:NTuple, ν0<:Real} <: AbstractSpectralModel
+#    """
+#    A tuple containing the Taylor expansion coefficients.
+#    c[1] is the spectral index α, c[2] is the spectral curvature β, etc.
+#    The coeffecients can be either constant values, or arrays with dimensions equal to that of the base image I0.
+#    """
+#    c::C
+#    """
+#    The expansion reference frequency.
+#    """
+#    ν0::C
+#end
+
+function order(::TaylorSpectral{N}) where {N}
+    return N
+end
+
+"""
+Applies Taylor Series spectral model to image data.
+
+Generates image data at frequency ν with respect to the reference frequency ν0.
+"""
+function applyspectral(I0::AbstractArray, spec::TaylorSpectral, ν::N) where {N<:Number}
+    data = copy(I0)
+    applyspectral!(data, spec, ν)
+    return data
+end
+
+function applyspectral!(I0::AbstractArray,
+                        spec::TaylorSpectral{<:Any,<:DataType,
+                                             <:Tuple{Vararg{<:AbstractArray}},<:Number,
+                                             <:Number}, ν::Number)
+    ν0 = spec.freq0
+    x = log(ν / ν0) # frequency to evaluate taylor expansion
+    c = spec.index
+
+    n = order(spec)
+    xlist = ntuple(i -> x^i, n) # creating a tuple to hold the frequency powers
+
+    for i in eachindex(I0) # doing expansion one pixel at a time
+        exparg = sum(getindex.(c, i) .* xlist)
+        I0[i] = I0[i] * exp(exparg)
+    end
+    return I0
+end
+
+function applyspectral!(I0::AbstractArray,
+                        spec::TaylorSpectral{<:Any,<:DataType,<:Tuple{Vararg{<:Number}},
+                                             <:Number,<:Number}, ν::Number)
+    ν0 = spec.freq0
+    x = log(ν / ν0) # frequency to evaluate taylor expansion
+    c = spec.index
+
+    n = order(spec)
+    xlist = ntuple(i -> x^i, n) # creating a tuple to hold the frequency powers
+
+    for i in eachindex(I0) # doing expansion one pixel at a time
+        exparg = sum(c .* xlist)
+        I0[i] = I0[i] * exp(exparg)
+    end
+    return I0
+end
+
+"""Given a multifrequency model (base image & spectral model), creates a new Continuous image model at frequency ν."""
+function generatemodel(MF::Multifrequency, ν::N) where {N<:Number}
+    image = parent(MF.base) # ContinuousImage -> SpatialIntensityMap
+    I0 = parent(image) # SpatialIntensityMap -> Array
+
+    data = applyspectral(I0, MF.spec, ν) # base image model, spectral model, frequency to generate new image
+
+    new_intensitymap = IntensityMap(data, getfield(image, :grid), getfield(image, :refdims),
+                                    getfield(image, :name))
```

Comment:
```suggestion
    new_intensitymap = rebuild(image; data=data)
```
is the recommended interface for this. Since `IntensityMap` is built off `DimensionalData` we can use their functions. This is explained in the [docs](https://rafaqz.github.io/DimensionalData.jl/stable/api/reference#DimensionalData.Dimensions.Lookups.rebuild). 

I really should add some developer documentation to Comrade to spell all of this out.